File: Part1_강의자료/Ch04_전체탐색_시뮬레이션/문제별코드/11068_회문인수/11068.cpp
#include <stdio.h>

int T, X;

int is_palindrome(int x, int base)
{
    int digit[20];
    int len = 0;

    while (x > 0)
    {
        digit[len++] = x % base;
        x /= base;
    }

    for (int i = 0; i < len / 2; i++)
        if (digit[i] != digit[len - i - 1])
            return 0;
    return 1;
}

int main()
{
    int ans;

    scanf("%d", &T);
    while (T--)
    {
        scanf("%d", &X);

        for (int i = 2; i <= 64; i++)
        {
            ans = is_palindrome(X, i);
            if (ans)
                break;
        }

        printf("%d\n", ans);
    }
    return 0;
}

File: Part1_강의자료/Ch04_전체탐색_시뮬레이션/문제별코드/11068_회문인수/11068.py
def is_palindrome(x, base):
    digit = []
    while x > 0:
        digit.append(x % base)
        x //= base

    for i in range(len(digit) // 2):
        if digit[i] != digit[-i - 1]:
            return 0
    return 1

T = int(input())

for _ in range(T):
    X = int(input())
    ans = 0

    for i in range(2, 65):
        ans = is_palindrome(X, i)
        if ans:
            break

    print(ans)
